src/ls/clients/regal.ts

```diff
@@ -337,30 +322,55 @@ function handleRegalShowEvalResult(params: any) {
         }
     }
 
-    // ruleNameDecorationOptions highlights only the rule name with a color.
-    // the colored highlight is displayed in addition to the whole line decoration
-    const ruleNameDecorationOptions: vscode.DecorationOptions = {
-        range: new vscode.Range(
-            new vscode.Position(line, 0),
-            new vscode.Position(line, ruleEnd),
-        ),
-    };
+    const ruleName = documentLine.text.substring(0, ruleEnd)
+
+    const decorationOptions: vscode.DecorationOptions[] = []
+    const ruleNameDecorationOptions: vscode.DecorationOptions[] = []
+
+    for (let line = 0; line < activeEditor.document.lineCount; line++) {
+        if (!activeEditor.document.lineAt(line).text.startsWith(ruleName)) {
```

Comment:
I guess the issue with this being in the client is things like:

```rego
import rego.v1


default allow := true

allow if false

allowed contains foo
```

Reply:
If we could return multiple results, for each rule from the backend based on the ast rule data: https://github.com/StyraInc/regal/blob/main/bundle/regal/ast/ast.rego#L95-L98 that'd be more robust, I think.